packages/cli-lib/api/dfs.js

```diff
@@ -40,11 +41,20 @@ async function createProject(portalId, name) {
  * @returns {Promise}
  */
 async function uploadProject(accountId, projectName, projectFile) {
+  const size = fs.lstatSync(projectFile).size;
+  const progressBar = new cliProgress.Bar();
+
+  let bytes = 0;
+
+  progressBar.start(size, 0);
+
   return http.post(accountId, {
     uri: `${PROJECTS_API_PATH}/upload/${projectName}`,
     timeout: 60000,
     formData: {
-      file: fs.createReadStream(projectFile),
+      file: fs.createReadStream(projectFile).on('data', chunk => {
+        progressBar.update((bytes += chunk.length));
```

Comment:
I think that you would want to track the progress of the POST request but I am not sure that the deprecated `request` npm module supports it based on discussion in https://github.com/request/request/issues/1450. Might be time to think about what it would take to start [shifting to another http library](https://github.com/HubSpot/hubspot-cli/issues/180). It looks like Axios [supports progress](https://github.com/axios/axios/blob/d99d5faac29899eba68ce671e6b3cbc9832e9ad8/examples/upload/index.html#L30-L32).

Reply:
Yeah, listening on the readable steam was the closest I could get with request.js. I did attempt to implement axios, thinking the same thing, but then I discovered that the `onUploadProgress` event is only supported in browsers.

<img width="600" alt="image" src="https://user-images.githubusercontent.com/9009552/132517542-73026f49-bf9d-4055-af81-7223601cdbd7.png">

Overall, the support for tracking POST/upload status in node is pretty poor from what I can tell. It does seem to be more straightforward with s3-style APIs.